mapie/conformity_scores/utils.py

```diff
@@ -1,26 +1,173 @@
+    (then the method have to be ``"lac"``), or multi-class.
 
     Parameters
     ----------
-    y_pred_proba: NDArray of shape (n_samples, n_classes)
-        Model prediction.
+    conformity_score: BaseClassificationScore
+        Conformity score function.
+
+    y: NDArray of shape (n_samples,)
+        Training labels.
+
+    Raises
+    ------
+    ValueError
+        If type of target is binary and method is not ``"lac"``
+        or ``"score"`` or if type of target is not multi-class.
+    """
+    check_classification_targets(y)
+    if type_of_target(y) == "binary" and not isinstance(conformity_score, LAC):
+        raise ValueError(
+            "Invalid method for binary target. "
+            "Your target is not of type multiclass and "
+            "allowed values for binary type are "
+            f"{['score', 'lac']}."
+        )
+
+
+METHOD_SCORE_MAP = {
+    'score': lambda: LAC(),
+    'lac': lambda: LAC(),
+    'cumulated_score': lambda: APS(),
+    'aps': lambda: APS(),
+    'naive': lambda: Naive(),
+    'raps': lambda: RAPS(),
+    'top_k': lambda: TopK()
+}
 
-    y: NDArray of shape (n_samples)
-        Labels.
 
-    Returns
-    -------
-    NDArray of shape (n_samples, 1)
-        Position of the true label in the prediction.
+def check_classification_conformity_score(
+    conformity_score: Optional[BaseClassificationScore] = None,
+    method: Optional[str] = None,
+) -> BaseClassificationScore:
     """
-    index = np.argsort(np.fliplr(np.argsort(y_pred_proba, axis=1)))
-    position = np.take_along_axis(index, y.reshape(-1, 1), axis=1)
+    Check parameter ``conformity_score`` for classification task.
 
-    return position
+    Raises
+    ------
+    ValueError
+        If parameters are not valid.
+
+    Examples
+    --------
+    >>> from mapie.conformity_scores.utils import (
+    ...     check_classification_conformity_score
+    ... )
+    >>> try:
+    ...     check_classification_conformity_score(1)
+    ... except Exception as exception:
+    ...     print(exception)
+    ...
+    Invalid conformity_score argument.
+    Must be None or a ConformityScore instance.
+    """
+    if method is None and conformity_score is None:
+        return LAC()
+    elif conformity_score is not None:
+        if method is not None:
+            warnings.warn(
+                "WARNING: the `conformity_score` parameter takes precedence "
+                "over the `method` parameter to define the method used.",
+                UserWarning
+            )
+        if isinstance(conformity_score, BaseClassificationScore):
+            return conformity_score
+    if method is not None:
+        if isinstance(method, str) and method in METHOD_SCORE_MAP:
+            _check_depreciated(method)
+            return METHOD_SCORE_MAP[method]()
+        else:
+            raise ValueError(
+                "Invalid method. "
+                f"Allowed values are {list(METHOD_SCORE_MAP.keys())}."
+            )
+    else:
```

Comment:
I think this else statement is at the wrong place as if it is not None and it's not an instance of BaseClassificaitonScore, then this raise is not accessed. Add a test to

Reply:
The test already exists and check if it is None, then if it is an instance of str, but not a correct string, then raise an error (see `test_check_wrong_classification_method` l101 and `test_check_conflict_parameters` l79 of `mapie/tests/test_conformity_scores_sets.py`).

Reply:
However, I forgot to raise an error when `conformity_score` is not of the right type, which I will add right away

Reply:
Done 👍